receiver/githubreceiver/trace_event_handling.go

```diff
@@ -130,3 +163,201 @@ func (gtr *githubTracesReceiver) createRootSpan(
+	}
+
+	jobSpanID, err := newJobSpanID(event.GetWorkflowJob().GetRunID(), int(event.GetWorkflowJob().GetRunAttempt()), event.GetWorkflowJob().GetName())
+	if err != nil {
+		return pcommon.SpanID{}, fmt.Errorf("failed to generate job span ID: %w", err)
+	}
+
+	span.SetTraceID(traceID)
+	span.SetParentSpanID(parentSpanID)
+	span.SetSpanID(jobSpanID)
+	span.SetName(event.GetWorkflowJob().GetName())
+	span.SetKind(ptrace.SpanKindServer)
+
+	// Workflow Job event start times provided by GitHub do not always match the
+	// start time of the actual job. Generally they are reported a second after
+	// the actual step start time. Thus, we use the first step start time as the
+	// span start time while using the normal completedAt time for the end time.
+	steps := event.GetWorkflowJob().Steps
+	if len(steps) > 0 {
+		span.SetStartTimestamp(pcommon.NewTimestampFromTime(steps[0].GetStartedAt().Time))
+		span.SetEndTimestamp(pcommon.NewTimestampFromTime(steps[len(steps)-1].GetCompletedAt().Time))
+	} else {
+		span.SetStartTimestamp(pcommon.NewTimestampFromTime(event.GetWorkflowJob().GetStartedAt().Time))
+		span.SetStartTimestamp(pcommon.NewTimestampFromTime(event.GetWorkflowJob().GetStartedAt().Time))
+	}
+
+	switch strings.ToLower(event.WorkflowJob.GetConclusion()) {
+	case "success":
+		span.Status().SetCode(ptrace.StatusCodeOk)
+	case "failure":
+		span.Status().SetCode(ptrace.StatusCodeError)
+	default:
+		span.Status().SetCode(ptrace.StatusCodeUnset)
+	}
+
+	span.Status().SetMessage(event.GetWorkflowJob().GetConclusion())
+
+	return parentSpanID, nil
+}
+
+// newJobSpanId creates a deterministic Job Span ID based on the provided runID,
+// runAttempt, and the name of the job.
+func newJobSpanID(runID int64, runAttempt int, jobName string) (pcommon.SpanID, error) {
+	input := fmt.Sprintf("%d%d%s", runID, runAttempt, jobName)
+	hash := sha256.Sum256([]byte(input))
+	spanIDHex := hex.EncodeToString(hash[:])
+
+	var spanID pcommon.SpanID
+	_, err := hex.Decode(spanID[:], []byte(spanIDHex[16:32]))
+	if err != nil {
+		return pcommon.SpanID{}, err
+	}
+
+	return spanID, nil
+}
+
+// createStepSpans is a wrapper function to create spans for each step in the
+// the workflow job by identifying duplicate names then creating a span for each
+// step.
+func (gtr *githubTracesReceiver) createStepSpans(
+	resourceSpans ptrace.ResourceSpans,
+	event *github.WorkflowJobEvent,
+	traceID pcommon.TraceID,
+	parentSpanID pcommon.SpanID,
+) error {
+	steps := event.GetWorkflowJob().Steps
+	unique := newUniqueSteps(steps)
+	var errors error
+	for i, step := range steps {
+		name := unique[i]
+		err := gtr.createStepSpan(resourceSpans, traceID, parentSpanID, event, step, name)
+		if err != nil {
+			errors = multierr.Append(errors, err)
+		}
+	}
+	return errors
+}
+
+// newUniqueSteps creates a new slice of step names from the provided GitHub
+// event steps. Each step name, if duplicated, is appended with `_n` where n is
```

Comment:
```suggestion
// event steps. Each step name, if duplicated, is appended with `-n` where n is
```
Nit, looks like it's actually a dash rather than underscore. (I don't have a strong preference which it should be.)

Reply:
Thanks for the catch.